bindings/src/main/java/io/pravega/storage/filesystem/FileSystemStorage.java

```diff
@@ -319,6 +319,7 @@ private Void doWrite(SegmentHandle handle, long offset, InputStream data, int le
                     totalBytesWritten += bytesWritten;
                     length -= bytesWritten;
                 }
+                channel.force(false);
```

Comment:
Actually, since you are in here, would you mind fixing the File size check at line 307?

We can check the file size and throw Bad Offset exception after we've opened the FileChannel by checking `FileChannel.size`.

At least this should reduce the number of IO calls a bit.

Reply:
sure.